src/core/QCheck2.ml

```diff
@@ -353,15 +361,20 @@ module Gen = struct
 
   let (--.) low high = float_range ?origin:None low high
 
-  let neg_int : int t = nat >|= Int.neg
+  let int_neg : int t = nat >|= Int.neg
 
-  (** [opt gen] shrinks towards [None] then towards shrinks of [gen]. *)
-  let opt ?(ratio : float = 0.85) (gen : 'a t) : 'a option t = fun st ->
+  let neg_int = int_neg
+
+  let option_ratio ~ratio (gen : 'a t) : 'a option t = fun st ->
     let p = RS.float st 1. in
     if p < (1. -. ratio)
     then Tree.pure None
     else Tree.opt (gen st)
 
+  let option gen = option_ratio ~ratio:0.85 gen
+
+  let opt ?(ratio = 0.85) = option_ratio ~ratio
```

Comment:
Does this compile without the dreaded "this-optional-argument-is-not-so-option" warning/error?
(I realize it is curried, so that `option_ratio` will expect a later `gen` argument)

Reply:
I did not have the warning but I will double check. But I think the `gen` argument should prevent the warning yes.